common/include/pcl/point_types_conversion.h

```diff
@@ -134,6 +136,58 @@ namespace pcl
     if (out.h < 0.f) out.h += 360.f;
   }
 
+  /** \brief Convert a XYZRGBA point type to a XYZLAB
+    * \param[in] in the input XYZRGBA point
+    * \param[out] out the output XYZLAB point
+    */
+  inline void
+  PointXYZRGBAtoXYZLAB (const PointXYZRGBA& in,
+                        PointXYZLAB&        out)
+  {
+    out.x = in.x;
+    out.y = in.y;
+    out.z = in.z;
+    out.data[3] = 1.0; // important for homogeneous coordinates
+
+    // convert sRGB to CIELAB
+    const Eigen::Vector3i colorRGB = in.getRGBVector3i();
+
+    // for sRGB   -> CIEXYZ see http://www.easyrgb.com/index.php?X=MATH&H=02#text2
+    // for CIEXYZ -> CIELAB see http://www.easyrgb.com/index.php?X=MATH&H=07#text7
+    // an overview at: https://www.comp.nus.edu.sg/~leowwk/papers/colordiff.pdf
+
+    const auto& sRGB_LUT = RGB2sRGB_LUT<double, 8>();
+
+    const double R = sRGB_LUT[colorRGB[0]];
+    const double G = sRGB_LUT[colorRGB[1]];
+    const double B = sRGB_LUT[colorRGB[2]];
+
+    // linear sRGB -> CIEXYZ, D65 illuminant, observer at 2 degrees
+    const double X = R * 0.4124 + G * 0.3576 + B * 0.1805;
+    const double Y = R * 0.2126 + G * 0.7152 + B * 0.0722;
+    const double Z = R * 0.0193 + G * 0.1192 + B * 0.9505;
+
+    // normalize X, Y, Z with tristimulus values for Xn, Yn, Zn
+    float f[3] = {static_cast<float>(X), static_cast<float>(Y), static_cast<float>(Z)};
+    f[0] /= 0.95047;
+    f[1] /= 1;
+    f[2] /= 1.08883;
+
+    // CIEXYZ -> CIELAB
+    for (int i = 0; i < 3; ++i) {
+      if (f[i] > 0.008856) {
+        f[i] = std::pow(f[i], 1.0 / 3.0);
+      }
+      else {
+        f[i] = 7.787 * f[i] + 16.0 / 116.0;
+      }
+    }
```

Comment:
I really really want to use the LUT here (but windows has higher error using the LUT than observed on linux). I tried, but failed in my PR. Do you happen to have access to a windows system?

Reply:
I haven't tried compile PCL in Windows, to be honest. But I'd have a try and get familiar with the Windows compile soon. Thank you for reminding me! I'll pay more attention to the cross-platform build in future.

Reply:
It's ok if you don't have access to windows. It's not a big issue :)